flocker/common/script.py

```diff
@@ -70,6 +71,42 @@ def main(reactor, options):
         :return: A ``Deferred`` which fires when the script has completed.
         """
 
+# This should probably be built-in functionality in Eliot;
+# see https://github.com/ClusterHQ/eliot/issues/143
+TWISTED_LOG_MESSAGE = MessageType("twisted:log",
+                                  fields(error=bool, message=unicode),
+                                  u"A log message from Twisted.")
+
+
+class EliotObserver(object):
+    """
+    A Twisted log observer that logs to Eliot.
+    """
+    def __init__(self, publisher=twisted_log):
+        """
+        :param publisher: A ``LogPublisher`` to capture logs from, or if no
+            argument is given the default Twisted log system.
+        """
+        self.logger = Logger()
+        self.publisher = publisher
+
+    def __call__(self, msg):
+        error = bool(msg.get("isError"))
+        message = unicode(textFromEventDict(msg), "charmap")
```

Comment:
I hadn't seen the use of "charmap" before. What's the purpose? Where can I read more about it?